test/endpoints.js

```diff
@@ -405,6 +405,52 @@ describe('Endpoints', function () {
         });
       });
 
+      describe('post versus from', function () {
```

Comment:
I'd also add a test for the negative condition: where `form` was given but does not match the form that was posted.

Similarly, where a `form` was given and it matches a subset of the post body. i.e., everything configured under `form` is present, but there were _extra_ things present in the posted form. That should still be a match.